features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt

```diff
@@ -291,6 +293,10 @@ internal open class RumViewScope(
             stopped = true
             sendViewUpdate(event, writer)
             sendViewChanged()
+
+            cpuVitalMonitor.unregister(cpuVitalListener)
+            memoryVitalMonitor.unregister(memoryVitalListener)
+            frameRateVitalMonitor.unregister(frameRateVitalListener)
```

Comment:
We should do the same also in `onStartView`, there is view stop logic there as well. And probably in `onStopSession`.